Libraries/Utilities/FileStoreExtensions.cs

```diff
@@ -0,0 +1,21 @@
+using System.Threading.Tasks;
+
+namespace OrchardCore.FileStorage
+{
+    public static class FileStoreExtensions
+    {
+        /// <summary>
+        /// Checks the existence of a file in the file store.
+        /// </summary>
+        /// <param name="path">Relative path to the file.</param>
+        public static async Task<bool> FileExistsAsync(this IFileStore fileStore, string path) =>
+            (await fileStore.GetFileInfoAsync(path)) != null;

```

Comment:
Lehetne `await` helyett `.ContinueWith(task => task.Result != null, TaskScheduler.Default)`, hogy ne hozzunk létre fölöslegesen dupla állapotgépet.

Reply:
De az randa :(.

Reply:
Illetve nem úgy tűnik, hogy a plusz állapotgép hiányával együtt az lenne az ajánlott: https://medium.com/@joni2nja/why-you-should-not-use-continuewith-in-your-async-code-c9eaf6087e64

Reply:
Okay then.